src/fs/fs.c

```diff
@@ -370,57 +375,24 @@ int cfs_fs_create(const str256_t *fs_name)
 	bufstat.st_mtim.tv_sec = 0;
 	bufstat.st_ctim.tv_sec = 0;
 
-	RC_WRAP_LABEL(rc, free_info, kvtree_create, ns, (void *)&bufstat,
-	              sizeof(struct stat), &kvtree);
-
-	/* Attach kvtree pointer to cortxfs struct */
-	fs_node->cfs_fs.kvtree = kvtree;
-
-	RC_WRAP_LABEL(rc, free_fs_node, kvtree_init, ns,
-		      fs_node->cfs_fs.kvtree);
+	RC_WRAP_LABEL(rc, delete_ns, kvtree_create, ns, (void *)&bufstat,
```

Comment:
I don't see anywhere you are using this `kvtree` after creating it. I believe you have to pass it inside `fs_node_init` or may be you can directly create the kvtree inside `fs_node_init`?

Reply:
I am not sure if we need to do that. I would like to file a new defect to study this code and check if ns and kvtree structures can be freed.

Reply:
Ok